### ISSUE TITLE
modernized libplot class

### ISSUE THREAD

**Issue by esheldon** [nolta/biggles#49]
I've made a branch "realobject" that modernizes the C extension, making a real python object of the libplot object.  This simplifies things a lot, in particular removing some race conditions and issues with the object getting destroyed.

THe C extension is also mostly python 3 compatible which should I hope mean some of the python3 stuff in another pull request may not have to be worried over.

Please test this branch and if it looks good we should merge it with master.  This should be done before merging the python3 branch.
